docs/src/architecture/09_architecture_decisions/0010-frontend-json-schema-constraints.md

```diff
@@ -0,0 +1,170 @@
+| `ProposalTemplate` | `summary.time.duration` | `integer` | Show the project duration in the UI components. The app needs to programmatically lookup in the template the related property. |
+| `ProposalTemplate` | `setup.proposer.applicant` | `string` | Show the proposal author in the UI components. The app needs to programmatically lookup in the template the related property. |
+| `ProposalTemplate` | `campaign_category` | `object` | App logic requires to show category information in the proposal builder but to hide it in the proposal viewer. The app needs to know which properties should not be rendered. This node and children nodes are excluded. |
+| `ProposalTemplate` | `campaign_category.category_details.details` | `object` | Show selected category details in the proposal builder. When the app sees this `nodeId` it will override the standard property widget with a category widget that fetches data externally and doesn't use the template property to render itself. The template has no data about the category. The app must know which property is a category property to know where in the proposal builder to put the overridden category widget. |
+| `ProposalTemplate` | `milestones.milestones` | `object` | Show the milestone count in the UI components. The app needs to programmatically lookup in the template the related property. |
+| `ProposalTemplate` | `milestones.milestones.milestone_list` | `array` | Show the milestone count in the UI components. The app needs to programmatically lookup in the template the related property. |
+| `ProposalTemplate` | `theme.theme.grouped_tag` | `object` | Show the proposal theme in the UI components. The app needs to programmatically lookup in the template the related property. |
+<!-- markdownlint-enable max-one-sentence-per-line -->
+
+## Schema Evolution Guidelines
+
+To safely evolve the schema over time while preserving functionality:
+
+1. **Never rename predefined `nodeId`s** - If structure needs to change, keep the old nodeId as an alias.
+  The app must lookup both the new nodeId but fallback to old nodeId if new one not found.
+2. **Use versioning** - Include schema version in document metadata
+3. **Add new fields rather than changing existing ones** - Mark old fields as deprecated
+4. **Test template changes** - Automated testing should verify that all predefined nodeIds are still accessible
+
+## Risks and Mitigations
+
+| Risk | Impact | Mitigation |
+|------|--------|------------|
+| Template modifications break nodeId lookups | UI shows placeholders instead of actual content | Create automated tests that verify all required nodeIds exist in template updates |
+| Schema grows too complex with custom logic | Maintenance burden increases | Document all custom behaviors and create a clear process for adding new ones |
+| Template authors don't understand constraints | Templates created that don't work with the app | Create a template authoring guide with clear examples and validation tools |
```

Comment:
This is why this should not be distinct from the proposal template specification and transitively the proposal meta template specification.  it is possible to enforce this using the proposal meta template, but only if it is clearly defined within the document specifications themselves.